model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/entities/CachedUser.java

```diff
@@ -119,4 +124,10 @@ public Set<String> getGroups(Supplier<UserModel> userModel) {
     public int getNotBefore() {
         return notBefore;
     }
+
+    public List<CredentialModel> getStoredCredentials(Supplier<UserModel> userModel) {
+        // TODO: should the list and its items be cloned first to avoid side effects?
```

Comment:
No, credential manipulation is done by special methods, and cloning `CredentialModel` is nontrivial.

Reply:
yes, clone it to be safe

Reply:
will be cloned in next commit